### DISCUSSION TITLE
Feature Request: eat support

### DISCUSSION THREAD

**Discussion by dpassen** [magit/with-editor#128] [Ideas]
eat (Emulate a Terminal) https://codeberg.org/akib/emacs-eat is a recently popular terminal emulator. It's comparable to vterm but doesn't require a native library nor a compilation step. It would be excellent if `with-editor` supported exporting to eat.

**Comment by brongulus** [comment]
Was going to create an issue for the same, with-editor doesn't export the variables in eat-mode, hope it can be supported!
Thank you tarsius for your wonderful project.

**Comment by sfllaw** [comment]
I’ve got a hack that works for me:
``` elisp
(cl-defun with-editor-export-editor-eat (process &optional (envvar "EDITOR"))
  "Like `with-editor-export-editor', but for `eat-exec-hook'."
  (cond
   ((derived-mode-p 'eat-mode)
    (if with-editor-emacsclient-executable
        (let ((with-editor--envvar envvar)
              (process-environment process-environment))
          (with-editor--setup)
          (while (accept-process-output process 0.1))
          (when-let ((v (getenv envvar)))
            (eat-term-send-string eat-terminal (format " export %s=%S" envvar v))
            (eat-self-input 1 'return))
          (when-let ((v (getenv "EMACS_SERVER_FILE")))
            (eat-term-send-string eat-terminal (format " export EMACS_SERVER_FILE=%S" v))
            (eat-self-input 1 'return))
          (eat-term-send-string eat-terminal "clear")
          (eat-self-input 1 'return))
      (error "Cannot use sleeping editor in this buffer")))
   (t (error "Cannot export environment variables in this buffer")))
  (message "Successfully exported %s" envvar))
(add-hook 'eat-exec-hook #'with-editor-export-editor-eat)
```

This is based on support for `vterm` in `with-editor-export-editor`. Note that `eat-exec-hook` is an _abnormal hook_ since it also passes down the `process` that drives the terminal, so `with-editor-export-editor-eat` accepts that extra argument.

**Comment by tarsius** [comment]
FYI, I plan to start using eat and teach with-editor about it. In that order. (Gotta do some other things first though.)

**Comment by wagk** [comment]
Heyo, wondering what's the current state of this? I'm trying to graft the two together in my config and can't seem to find much guidance anywhere.

**Comment by tarsius** [comment]
Implemented in ac561dc0c5e3108e2f693d17569dd1fae3847e97.

**Reply by sfllaw** [reply]
@tarsius, amazing! Thanks.

**Reply by tarsius** [reply]
You are welcome!

I extended `with-editor-export-editor` so that `with-editor-export-git-editor` et al. can also be used (as commands).

**Reply by tarsius** [reply]
> ... can also be used (as commands).

Which made me realize I should do b48b586bb175fcdfea0a3372ee5a4dc12eac2e99 and 2671aa47e610cbb2763858507d881bdb90b1065a.
